File: frontend/java-retrofit/app/src/main/java/ma/ensa/reservation/ui/chambres/AddChambreActivity.java
package ma.ensa.reservation.ui.chambres;

import android.os.Bundle;
import android.widget.ArrayAdapter;
import android.widget.Button;
import android.widget.EditText;
import android.widget.Spinner;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import ma.ensa.reservation.R;
import ma.ensa.reservation.api.ApiClient;
import ma.ensa.reservation.api.ApiInterface;
import ma.ensa.reservation.models.Chambre;
import ma.ensa.reservation.models.DispoChambre;
import ma.ensa.reservation.models.TypeChambre;
import retrofit2.Call;
import retrofit2.Callback;
import retrofit2.Response;

public class AddChambreActivity extends AppCompatActivity {

    private Spinner spinnerTypeChambre, spinnerDispoChambre;
    private EditText editPrix;
    private Button btnSave;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_add_chambre);

        // Initialisation des vues
        spinnerTypeChambre = findViewById(R.id.spinnerTypeChambre);
        editPrix = findViewById(R.id.editPrix);
        spinnerDispoChambre = findViewById(R.id.spinnerDispoChambre);
        btnSave = findViewById(R.id.btnSave);

        // Configurer les Spinners
        setupSpinners();

        // Bouton de sauvegarde
        btnSave.setOnClickListener(v -> saveChambre());
    }

    private void setupSpinners() {
        // Adapter pour TypeChambre
        ArrayAdapter<TypeChambre> typeChambreAdapter = new ArrayAdapter<>(
                this, android.R.layout.simple_spinner_item, TypeChambre.values());
        typeChambreAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinnerTypeChambre.setAdapter(typeChambreAdapter);

        // Adapter pour DispoChambre
        ArrayAdapter<DispoChambre> dispoChambreAdapter = new ArrayAdapter<>(
                this, android.R.layout.simple_spinner_item, DispoChambre.values());
        dispoChambreAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item);
        spinnerDispoChambre.setAdapter(dispoChambreAdapter);
    }

    private void saveChambre() {
        String prixStr = editPrix.getText().toString();

        // Validation des champs
        if (prixStr.isEmpty()) {
            Toast.makeText(this, "Veuillez remplir tous les champs", Toast.LENGTH_SHORT).show();
            return;
        }

        try {
            // Créer une nouvelle chambre
            Chambre chambre = new Chambre();
            chambre.setTypeChambre((TypeChambre) spinnerTypeChambre.getSelectedItem());
            chambre.setPrix(Double.parseDouble(prixStr));
            chambre.setDispoChambre((DispoChambre) spinnerDispoChambre.getSelectedItem());

            // Envoyer la chambre à l'API
            ApiInterface apiService = ApiClient.getClient().create(ApiInterface.class);
            Call<Chambre> call = apiService.createChambre(chambre);
            call.enqueue(new Callback<Chambre>() {
                @Override
                public void onResponse(Call<Chambre> call, Response<Chambre> response) {
                    if (response.isSuccessful()) {
                        Toast.makeText(AddChambreActivity.this, "Chambre ajoutée", Toast.LENGTH_SHORT).show();
                        setResult(RESULT_OK);
                        finish(); // Fermer l'activité après l'ajout
                    } else {
                        Toast.makeText(AddChambreActivity.this, "Erreur d'ajout", Toast.LENGTH_SHORT).show();
                    }
                }

                @Override
                public void onFailure(Call<Chambre> call, Throwable t) {
                    Toast.makeText(AddChambreActivity.this, "Erreur de connexion", Toast.LENGTH_SHORT).show();
                }
            });
        } catch (NumberFormatException e) {
            Toast.makeText(this, "Prix invalide", Toast.LENGTH_SHORT).show();
        }
    }
}

File: frontend/kotlin-retrofit/app/src/main/java/ma/ensa/reservationretrofitko/ui/chambres/AddChambreActivity.kt
package ma.ensa.reservationretrofitko.ui.chambres

import android.os.Bundle
import android.widget.ArrayAdapter
import android.widget.Button
import android.widget.EditText
import android.widget.Spinner
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity
import ma.ensa.reservationretrofitko.R
import ma.ensa.reservationretrofitko.api.ApiClient
import ma.ensa.reservationretrofitko.api.ApiInterface
import ma.ensa.reservationretrofitko.models.Chambre
import ma.ensa.reservationretrofitko.models.DispoChambre
import ma.ensa.reservationretrofitko.models.TypeChambre
import retrofit2.Call
import retrofit2.Callback
import retrofit2.Response

class AddChambreActivity : AppCompatActivity() {

    private lateinit var spinnerTypeChambre: Spinner
    private lateinit var editPrix: EditText
    private lateinit var spinnerDispoChambre: Spinner
    private lateinit var btnSave: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_add_chambre)

        // Initialisation des vues
        spinnerTypeChambre = findViewById(R.id.spinnerTypeChambre)
        editPrix = findViewById(R.id.editPrix)
        spinnerDispoChambre = findViewById(R.id.spinnerDispoChambre)
        btnSave = findViewById(R.id.btnSave)

        // Configurer les Spinners
        setupSpinners()

        // Bouton de sauvegarde
        btnSave.setOnClickListener { saveChambre() }
    }

    private fun setupSpinners() {
        // Adapter pour TypeChambre
        val typeChambreAdapter = ArrayAdapter(
            this, android.R.layout.simple_spinner_item, TypeChambre.values()
        )
        typeChambreAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinnerTypeChambre.adapter = typeChambreAdapter

        // Adapter pour DispoChambre
        val dispoChambreAdapter = ArrayAdapter(
            this, android.R.layout.simple_spinner_item, DispoChambre.values()
        )
        dispoChambreAdapter.setDropDownViewResource(android.R.layout.simple_spinner_dropdown_item)
        spinnerDispoChambre.adapter = dispoChambreAdapter
    }

    private fun saveChambre() {
        val prixStr = editPrix.text.toString()

        // Validation des champs
        if (prixStr.isEmpty()) {
            Toast.makeText(this, "Veuillez remplir tous les champs", Toast.LENGTH_SHORT).show()
            return
        }

        try {
            // Créer une nouvelle chambre
            val chambre = Chambre().apply {
                typeChambre = spinnerTypeChambre.selectedItem as TypeChambre
                prix = prixStr.toDouble()
                dispoChambre = spinnerDispoChambre.selectedItem as DispoChambre
            }

            // Envoyer la chambre à l'API
            val apiService = ApiClient.getClient().create(ApiInterface::class.java)
            val call = apiService.createChambre(chambre)
            call.enqueue(object : Callback<Chambre> {
                override fun onResponse(call: Call<Chambre>, response: Response<Chambre>) {
                    if (response.isSuccessful) {
                        Toast.makeText(this@AddChambreActivity, "Chambre ajoutée", Toast.LENGTH_SHORT).show()
                        setResult(RESULT_OK)
                        finish() // Fermer l'activité après l'ajout
                    } else {
                        Toast.makeText(this@AddChambreActivity, "Erreur d'ajout", Toast.LENGTH_SHORT).show()
                    }
                }

                override fun onFailure(call: Call<Chambre>, t: Throwable) {
                    Toast.makeText(this@AddChambreActivity, "Erreur de connexion", Toast.LENGTH_SHORT).show()
                }
            })
        } catch (e: NumberFormatException) {
            Toast.makeText(this, "Prix invalide", Toast.LENGTH_SHORT).show()
            }
        }
}